pyronear_ds/datasets/utils.py

```diff
@@ -1,18 +1,198 @@
+        with open(full_path, 'wb+') as file:
+            file.write(content)
+
+    elif not unzip and isinstance(compression, str):
+        content = url_retrieve(url)
+        file_name = f"{base}.{compression}"
+        full_path = os.path.join(destination, file_name)
+        os.makedirs(os.path.dirname(full_path), exist_ok=True)
+        with open(full_path, 'wb+') as file:
+            file.write(content)
+
+    else:
+        raise ValueError("If the file is not compressed set unzip to False")
+
+
+def get_ghcn(start_year: Optional[int] = None, end_year: Optional[int] = None, destination: Optional[str] = './ghcn'):
+    """Download yearly Global Historical Climatology Network - Daily (GHCN-Daily) (.csv) From (NCEI).
+
+    Args:
+        start_year: first year to be retrieved. Defaults to None.
+        end_year: first that will not be retrieved. Defaults to None.
+        destination: destination directory. Defaults to './ghcn'.
+    """
+# TODO
+# Write case tests
+# Implement archive=False
+    start_year = datetime.now().year if start_year is None else start_year
+    end_year = datetime.now().year + 1 if end_year is None or start_year == end_year else end_year
+
+    for year in range(start_year, end_year):
+        url = f"https://www1.ncdc.noaa.gov/pub/data/ghcn/daily/by_year/{year}.csv.gz"
+        download(url=url, default_extension='csv', unzip=True, destination=destination)
+
+
+def get_modis(start_year: Optional[int] = None, end_year: Optional[int] = None, yearly: Optional[bool] = False,
+              destination: Optional[str] = './firms'):
+    """Download last 24H or yearly France active fires from the FIRMS NASA.
+
+    Args:
+        start_year: first year to be retrieved. Defaults to None.
+        end_year: first that will not be retrieved. Defaults to None.
+        yearly: whether to download yearly active fires or not. Defaults to False.
+        destination: destination directory. Defaults to './firms'.]
+    """
+    if yearly is True:
+        start_year = datetime.now().year - 1 if start_year is None else start_year
+        end_year = datetime.now().year if end_year is None or start_year == end_year else end_year
+
+        for year in range(start_year, end_year):
+            assert (start_year != 2020 or end_year != 2021), \
+                'MODIS active fire archives are only available for the years from 2000 to 2019'
+            url = f"https://firms.modaps.eosdis.nasa.gov/data/country/modis/{year}/modis_{year}_France.csv"
+            download(url=url, default_extension='csv', unzip=False, destination=destination)
+
+    else:
+        if start_year is not None:
+            raise warnings.warn("The active fires from the last 24H of the MODIS Satellite will be download.")
+        else:
+            url = "https://firms.modaps.eosdis.nasa.gov/data/active_fire/c6/csv/MODIS_C6_Europe_24h.csv"
+            download(url=url, default_extension='csv', unzip=False, destination=destination)
```

Comment:
for later : those two functions should be moved to methods of the corresponding dataset classes once we have created them!